src/github/summary.ts

```diff
@@ -0,0 +1,96 @@
+import {context} from '@actions/github'
+import {intervalToDuration} from 'date-fns'
+import {bold, emoji, link} from '../slack/mrkdwn'
+import {Link} from '../slack/types'
+import {dateFromTs} from '../slack/utils'
+import {getContextBlock} from './context'
+import {createMessage, emojiFromStatus} from './message'
+import {JobStatus, Message, Text} from './types'
+import {isPullRequestEvent, isPushEvent} from './webhook'
+
+interface Options {
+  status: string
+  threadTs: string
+  now: Date
+}
+
+/**
+ * Return the initial summary message.
+ */
+export function getSummaryMessage(options?: Options): Message {
+  const text = getText(options?.status)
+
+  const duration = options
+    ? intervalToDuration({
+        start: dateFromTs(options.threadTs),
+        end: options.now
+      })
+    : undefined
+
+  const contextBlock = getContextBlock(duration)
+
+  return createMessage(text, contextBlock)
+}
+
+function getText(status?: string): Text {
+  const summarySentence = getSummarySentence(status)
+  const eventLink = getEventLink()
+
+  const mrkdwn = [
+    status ? emojiFromStatus(status) : emoji('black_square_button'),
```

Comment:
I don't personally love the initial / in progress / "unchecked" icon:

<img width="453" alt="Screen Shot 2022-09-11 at 3 22 59 PM" src="https://user-images.githubusercontent.com/847532/189545392-c1a29b55-6cb7-4bf1-84a8-6e7bc2f49a1b.png">

Perhaps we make this configurable at some point.

Reply:
That'd be sweet! Big fan of using custom emojis :D

Reply:
Tracking via https://github.com/namoscato/action-slack-deploy-pipeline/issues/13.